blockly-rtc/server/http_server.js

```diff
@@ -34,6 +34,12 @@ http.createServer(async (req, res) => {
     await queryEventsHandler_(res, parsedUrl.query.serverId);
   } else if (req.method === 'POST' && parsedUrl.pathname === '/api/events/add') {
     await addEventsHandler_(req, res);
+  } else if (req.method === 'GET' && parsedUrl.pathname === '/api/markers/query') {
+    await getMarkerUpdatesHandler_(res, parsedUrl.query.workspaceId);
+  } else if (req.method === 'POST' && parsedUrl.pathname === '/api/markers/add') {
+    await addClientHandler_(res, parsedUrl.query.workspaceId);
```

Comment:
You can combine the add and update actions by using an [UPSERT sql command](https://www.sqlite.org/lang_UPSERT.html) and only sending updates. This will also remove the need for clients to keep track of their registered state on the server or having repeated adds for the same client.

Reply:
This is amazing! Thanks you for pointing me towards this.